File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def __init__(self):
        self.vec = []

    def inorder(self, node):
        if node is None:
            return
        self.inorder(node.left)
        self.vec.append(node.val)
        self.inorder(node.right)

    def getMinimumDifference(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        self.inorder(root)
        ans = float('inf')
        for i in range(1, len(self.vec)):
            ans = min(ans, abs(self.vec[i] - self.vec[i-1]))
        return ans
        

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
private:
    int result_ = INT_MAX;
    TreeNode* prev_ = nullptr;

public:
    void inorder(TreeNode* node) {
        if (!node)
            return;
        inorder(node->left);
        if (prev_)
            result_ = min(result_, abs(prev_->val - node->val));
        prev_ = node;
        inorder(node->right);
    }

    int getMinimumDifference(TreeNode* root) {
        inorder(root);
        return result_;
    }
};